bin/scripts/continious-data-population.js

```diff
@@ -0,0 +1,1861 @@
+
+        this.ts += 1000;
+
+        event.segmentation = {};
+        event.segmentation.type = "click";
+
+        var dice = getRandomInt(0, 6) % 2 === 0 ? true : false;
+        if (dice) {
+            var randomIndex = getRandomInt(0, selectedOffsets.length - 1);
+            event.segmentation.x = selectedOffsets[randomIndex].x;
+            event.segmentation.y = selectedOffsets[randomIndex].y;
+        }
+        else {
+            event.segmentation.x = getRandomInt(0, 1440);
+            event.segmentation.y = getRandomInt(0, 990);
+        }
+
+        event.segmentation.width = 1440;
+        event.segmentation.height = 3586;
+        event.segmentation.domain = SERVER_URL;
+        event.segmentation.view = views[Math.floor(Math.random() * views.length)];
+        return [event];
+    };
+
+    this.getScrollmapEvents = function(appKey, templateId) {
+        var events = this.getHeatmapEvent(appKey, templateId);
+
+        if (Math.random() >= 0.5) {
+            events = events.concat(this.getScrollmapEvent(appKey, templateId));
+
+            if (Math.random() >= 0.8) {
+                events = events.concat(this.getScrollmapEvent(appKey, templateId));
+            }
+        }
+
+        return events;
+    };
+
+    this.getScrollmapEvent = function(appKey, templateId) {
+        var populatorType = null;
+        if (templateId.length <= 10) { //defaultTemplate
+            defaultTemplates.forEach(function(template) {
+                if (template.name === templateId) {
+                    populatorType = template._id;
+                }
+            });
+        }
+        else {
+            populatorType = template._id;
+        }
+        var views = ["/populator/" + appKey + "/demo-" + populatorType + ".html"];
+        var event = {
+            "key": "[CLY]_action",
+            "count": 1,
+            "timestamp": this.ts,
+            "hour": getRandomInt(0, 23),
+            "dow": getRandomInt(0, 6),
+            "test": 1
+        };
+        this.ts += 1000;
+        event.segmentation = {};
+        event.segmentation.type = "scroll";
+        event.segmentation.y = getRandomInt(0, 3602) + 983;
+        event.segmentation.width = 1440;
+        event.segmentation.height = 3586;
+        event.domain = SERVER_URL;
+        event.segmentation.view = views[Math.floor(Math.random() * views.length)];
+        return [event];
+    };
+
+    this.addEvent = function(template, app, templateId) {
+        var req = {};
+        var events;
+
+        if (!this.isRegistered) {
+            this.isRegistered = true;
+            events = this.getEvent("[CLY]_view", template && template.events && template.events["[CLY]_view"], app.key, templateId).concat(this.getEvent("[CLY]_orientation", template && template.events && template.events["[CLY]_orientation"], app.key, templateId), this.getEvents(4, template && template.events, app.key, templateId));
```

Comment:
I thought having one event here.
this.getEvents(1, template && template.events, app.key, templateId)